src/Adapter/Product/QueryHandler/GetProductCombinationsForEditingHandler.php

```diff
@@ -0,0 +1,149 @@
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(GetProductCombinationsForEditing $query): CombinationListForEditing
+    {
+        $productId = $query->getProductId();
+        $this->productRepository->assertProductExists($productId);
+        $combinations = $this->combinationRepository->getProductCombinations($productId, $query->getLimit(), $query->getOffset());
+
+        $combinationIds = array_map(function ($combination): int {
+            return (int) $combination['id_product_attribute'];
+        }, $combinations);
+
+        $attributesInformation = $this->combinationRepository->getAttributesInfoByCombinationIds(
+            $combinationIds,
+            $query->getLanguageId()
+        );
+
+        return $this->formatCombinationsForEditing(
+            $combinations,
+            $attributesInformation,
+            $this->combinationRepository->getTotalCombinationsCount($productId)
+        );
+    }
+
+    /**
+     * @param array $combinations
+     * @param array<int, array<int, mixed>> $attributesInformationByCombinationId
+     * @param int $totalCombinationsCount
+     *
+     * @return CombinationListForEditing
+     */
+    private function formatCombinationsForEditing(
+        array $combinations,
+        array $attributesInformationByCombinationId,
+        int $totalCombinationsCount
+    ): CombinationListForEditing {
+        $combinationsForEditing = [];
+
+        foreach ($combinations as $combination) {
+            $combinationId = (int) $combination['id_product_attribute'];
+            $combinationAttributesInformation = [];
+
+            foreach ($attributesInformationByCombinationId[$combinationId] as $attributesInfo) {
+                foreach ($attributesInfo as $attributeInfo) {
+                    $combinationAttributesInformation[] = new CombinationAttributeInformation(
+                        (int) $attributeInfo['id_attribute_group'],
+                        $attributeInfo['attribute_group_name'],
+                        (int) $attributeInfo['id_attribute'],
+                        $attributeInfo['attribute_name']
+                    );
+                }
+            }
+
+            $combinationsForEditing[] = new CombinationForEditing(
+                $combinationId,
+                $this->buildCombinationName($combinationAttributesInformation),
+                $combinationAttributesInformation
+            );
+        }
+
+        return new CombinationListForEditing($totalCombinationsCount, $combinationsForEditing);
+    }
+
+    /**
+     * @param CombinationAttributeInformation[] $attributesInformation
+     *
+     * @return string
+     */
+    private function buildCombinationName(array $attributesInformation): string
+    {
+        $combinedNameParts = [];
+        foreach ($attributesInformation as $combinationAttributeInformation) {
+            $combinedNameParts[] = sprintf(
+                '%s - %s',
```

Comment:
```suggestion
                '%s - %s',
```

Please change the ` - ` by the configuration option : `PS_ATTRIBUTE_ANCHOR_SEPARATOR`